tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -166,14 +166,18 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
         """ General data processing steps used for all models. """
         X = copy.deepcopy(X)
 
-        # Remove all examples with missing labels from this dataset:
-        missinglabel_inds = list(X[X[self.label].isna()].index)
-        if len(missinglabel_inds) > 0:
-            logger.warning(f"Warning: Ignoring {len(missinglabel_inds)} (out of {len(X)}) training examples for which the label value in column '{self.label}' is missing")
-            X = X.drop(missinglabel_inds, axis=0)
+        if pd.api.types.is_float_dtype(X[self.label]):
+            has_invalid_labels = np.any(~np.isfinite(X[self.label]))
+            if has_invalid_labels:
+                raise ValueError("Label column cannot contain non-finite values (NaN, Inf, Ninf)")
```

Comment:
[minor] for the convenience of the user to debug the issue in the label. can we add one line here to indicate the first index of the dataset which has non supported values?